RecoEgamma/EgammaTools/src/EgammaPCAHelper.cc

```diff
@@ -19,18 +19,13 @@ EGammaPCAHelper::EGammaPCAHelper()
       // See RecoLocalCalo.HGCalRecProducers.HGCalRecHit_cfi
       invThicknessCorrection_({1. / 1.132, 1. / 1.092, 1. / 1.084}),
       pca_(new TPrincipal(3, "D")) {
-  hitMapOrigin_ = 0;
-  hitMap_ = new std::map<DetId, const HGCRecHit*>();
+  hitMap_ = new std::unordered_map<DetId, const HGCRecHit*>();
```

Comment:
This will cause a memory leak. This instance of an object is never supposed to be used so it seems you should just be able to do 
```suggestion
  hitMap_ = nullptr;
```

By the way, is `pca_` also going to cause a memory leak? What code is responsible for deleting it?

Reply:
I should have checked further, `pca_` is a `std::unique_ptr` so the memory is properly managed.